staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1140,7 +1156,15 @@ func (lw *cacherListerWatcher) Watch(options metav1.ListOptions) (watch.Interfac
 	if utilfeature.DefaultFeatureGate.Enabled(features.EfficientWatchResumption) {
 		opts.ProgressNotify = true
 	}
-	return lw.storage.WatchList(context.TODO(), lw.resourcePrefix, opts)
+	return lw.storage.WatchList(lw.createKCPClusterAwareContext(), lw.resourcePrefix, opts)
+}
+
+func (lw *cacherListerWatcher) createKCPClusterAwareContext() context.Context {
```

Comment:
nit: `func createKCPClusterAwareContext(meta *storagebackend.KcpStorageMetadata)`.

rule-of-thumb: functional (= non-side-effect) funcs should not be methods.